### PR TITLE
Dont insert reshapes when converting pooling to reduce

### PR DESCRIPTION
**Pull request by pfultz2**
This should improve fusions.

### CODE REVIEW

**Comment by umangyadav** [comment]
Performance tests have compilation error

**Comment by umangyadav** [comment]
@pfultz2  accuracy tests are still failing during compile. Can you check ?

**Comment by pfultz2** [comment]
> @pfultz2 accuracy tests are still failing during compile. Can you check ?

@umangyadav I pushed a fix.

File: src/rewrite_pooling.cpp [hunk 1]
```diff
@@ -54,27 +52,18 @@ void rewrite_pooling::apply(module& m) const
...
         if(not std::equal(lens.begin() + 2, lens.end(), op.lengths.begin(), op.lengths.end()))
             continue;
-        std::int64_t n = s.lens()[0];
-        std::int64_t c = s.lens()[1];
-        auto reshape   = m.insert_instruction(
-            ins, make_op("reshape", {{"dims", {n * c, -1}}}), ins->inputs().front());
-        instruction_ref pooling{};
-
+        std::vector<std::int64_t> axes(lens.size() - 2);
+        std::iota(axes.begin(), axes.end(), 2);
```
Comment (umangyadav) [inline src/rewrite_pooling.cpp:56]: Should this be doing any checks for the Layout ?

File: src/simplify_reshapes.cpp [hunk 2]
```diff
@@ -122,6 +122,11 @@ struct find_nop_reshapes
         reshapes.insert("pad");
         reshapes.insert("slice");
         reshapes.insert("transpose");
+        reshapes.insert("reduce_mean");
```
Comment (umangyadav) [inline src/simplify_reshapes.cpp:125]: Would be better to assert that axes are either empty resulting in no-op or 
axes  lengths are 1.

**Comment by umangyadav** [comment]
> This should improve fusions. 

is it improving reduce fusion ? Which of the matcher is helping doing that ?

**Review by umangyadav — APPROVED** [review]

File: src/rewrite_pooling.cpp (hunk 1 above)
Comment (umangyadav) [reply src/rewrite_pooling.cpp:56]: @pfultz2  sorry i missed this comment before approving. Would this require check on the layout ?

**Review by TedThemistokleous — APPROVED** [review]

**Pull request merged**